### DISCUSSION TITLE
How to manually change data to null when useQuery fails

### DISCUSSION THREAD

**Discussion by unicape** [TanStack/query#6030] [Q&A]
How to manually change data to null when useQuery fails

```
const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        keepPreviousData: true,
        retry: false,
      },
    },
  })
```

**Comment by clicktodev** [comment]
Something like this should do what you're looking for on a query level
Mark as an answer if it solves your problem
```js
export const useGetCurrentUser = () => {
	const queryClient = useQueryClient();

	return useQuery(
		["current-user"],
		async () => {
			const res = await axiosClient.get(`/user`);
			return res.data;
		},
		{
			onError: () => {
				queryClient.setQueryData(["current-user"], (old) => null);
			},
		}
	);
};

```

**Reply by TkDodo** [reply]
`onError` is deprecated. What you want is to catch the error in the queryFn itself and return null

**Reply by clicktodev** [reply]
Good to know, i didn't realize it was deprecated in v5.
The active stable is v4 so my code should work for @unicape

**Reply by manasb-uoe** [reply]
what could we do in v5 and beyond? @hichemfantar
